core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala

```diff
@@ -68,17 +68,28 @@ trait H2OContextRestAPIUtils extends H2OContextUtils  {
   }
 
   private def getCloudInfoFromNode(endpoint: URI): CloudV3 = {
-    val content = readURLContent(s"$endpoint/3/Cloud")
+    val content = readURLContent(endpoint, "3/Cloud")
     new Gson().fromJson(content, classOf[CloudV3])
   }
 
-  private def readURLContent(url: String): String = {
-    val html = Source.fromURL(url)
-    val content = html.mkString
-    html.close()
-    content
+  private def readURLContent(endpoint: URI, suffix: String): String = {
+    try {
+      val html = Source.fromURL(s"$endpoint/$suffix")
+      val content = html.mkString
+      html.close()
+      content
+    } catch {
+      case cause: Exception =>
+        H2OContext.get().foreach(_.stop())
```

Comment:
I'm wondering whether this class is right place for stopping `H2OContext`. This class looks after commonication with a rest endpoint. Do you think we could find a different place for `H2OContext.get().foreach(_.stop())`?  WDYT?

Reply:
Yes, agree, please have a look now.

Right now we just throw exception if the node is not reachable. The only place where cluster is killed is inside the thread getting heartbeats from the cluster (might be good refactoring to rename this thread a little bit, but I wanted to keep this PR as small as possible)